### ISSUE TITLE
Support container level security context

### ISSUE THREAD

**Issue by zamsong123** [artemiscloud/activemq-artemis-operator#683]
Is your feature request related to a problem? Please describe.
Currently our EKS environment have some pre check on the container level security context, deployment will be blocked if not set

Describe the solution you'd like
Add a field to specify container level security context

**Comment by zamsong123** [comment]
will make changes and create PR to this issue

**Comment by brusdev** [comment]
@zamsong123 the containers should inherit the pod level security context, are you setting the field `spec.deploymentPlan.podSecurityContext` in your ActiveMQArtemis CR?

**Comment by zamsong123** [comment]
@brusdev  thanks for your comments, in our case, our EKS supporting team is doing some restriction on this, e.g. if there's no container level "runAsRootUser": true is set, the deployment will be blocked, frankly speaking I am not sure how the inherit works, but I guess they are different configuration for different purpose

**Comment by zamsong123** [comment]
@brusdev yes I tried the spec.deploymentPlan.podSecurityContext, it's not working for our EKS requirement

**Comment by zamsong123** [comment]
@brusdev  btw, in Confluent Operator, they are also separate config for different purpose

**Comment by brusdev** [comment]
@zamsong123 thanks for the clarification, your enhancement makes sense to me

**Comment by Eric84626** [comment]
I will make code change for this issue.

**Comment by brusdev** [comment]
@Eric84626 cool, contributions are always welcome!

**Comment by brusdev** [comment]
This issue is incorporated by #758
